### DISCUSSION TITLE
ItemFrame + Map + MapMeta

### DISCUSSION THREAD

**Discussion by PlugDev-class** [Minestom/Minestom#1129] [Help] [answered]
Simple idea:

I want to have a Map with a picture in it.
Then I write something like a string on it.
Spawn an ItemFrame to a specific position (Pos)
Set the Map with MapMeta in the ItemFrame..

I'm literally overwhelmed to spawn an ItemFrame with a Map (MapMeta) in it.
The generation of the map is no problem. I use java.awt.Graphics2d for rendering.

**Comment by KrystilizeNevaDies** [comment, marked as answer]
It should be something along the lines of this:
```java

// You first need to generate the map data packet with your map id.
// Then send this map data packet to all players

ItemStack mapItem = ItemStack.builder(Material.Map)
	.meta(MapMeta.class, meta -> meta.setMapID(yourMapID)
	.build();

Entity itemFrame = new Entity(EntityType.ItemFrame);
ItemFrameMeta meta = (ItemFrameMeta) itemFrame.getMeta();
meta.setItemStack(mapItem);

// Now you can use ItemFrame in your instance
```
(I apologise if I mistyped anything, don't have time to check manually)

**Reply by PlugDev-class** [reply]
Thank you so much!
